docs/core_docs/docs/tutorials/rag.ipynb

```diff
@@ -1,831 +1,1202 @@
+    "complexity.\n",
+    "\n",
+    "If you're already familiar with basic retrieval, you might also be interested in\n",
+    "this [high-level overview of different retrieval techinques](/docs/concepts/retrieval).\n",
+    "\n",
+    "**Note**: Here we focus on Q&A for unstructured data. If you are interested for RAG over structured data, check out our tutorial on doing [question/answering over SQL data](/docs/tutorials/sql_qa).\n",
+    "\n",
+    "## Overview\n",
+    "A typical RAG application has two main components:\n",
+    "\n",
+    "**Indexing**: a pipeline for ingesting data from a source and indexing it. *This usually happens offline.*\n",
+    "\n",
+    "**Retrieval and generation**: the actual RAG chain, which takes the user query at run time and retrieves the relevant data from the index, then passes that to the model.\n",
+    "\n",
+    "Note: the indexing portion of this tutorial will largely follow the [semantic search tutorial](/docs/tutorials/retrievers).\n",
+    "\n",
+    "The most common full sequence from raw data to answer looks like:\n",
+    "\n",
+    "### Indexing\n",
+    "1. **Load**: First we need to load our data. This is done with [Document Loaders](/docs/concepts/document_loaders).\n",
+    "2. **Split**: [Text splitters](/docs/concepts/text_splitters) break large `Documents` into smaller chunks. This is useful both for indexing data and passing it into a model, as large chunks are harder to search over and won't fit in a model's finite context window.\n",
+    "3. **Store**: We need somewhere to store and index our splits, so that they can be searched over later. This is often done using a [VectorStore](/docs/concepts/vectorstores) and [Embeddings](/docs/concepts/embedding_models) model.\n",
+    "\n",
+    "![index_diagram](../../static/img/rag_indexing.png)\n",
+    "\n",
+    "### Retrieval and generation\n",
+    "4. **Retrieve**: Given a user input, relevant splits are retrieved from storage using a [Retriever](/docs/concepts/retrievers).\n",
+    "5. **Generate**: A [ChatModel](/docs/concepts/chat_models) / [LLM](/docs/concepts/text_llms) produces an answer using a prompt that includes both the question with the retrieved data\n",
+    "\n",
+    "![retrieval_diagram](../../static/img/rag_retrieval_generation.png)\n",
+    "\n",
+    "Once we've indexed our data, we will use [LangGraph](https://langchain-ai.github.io/langgraphjs/) as our orchestration framework to implement the retrieval and generation steps.\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "### Jupyter Notebook\n",
+    "\n",
+    "This and other tutorials are perhaps most conveniently run in a [Jupyter notebooks](https://jupyter.org/). Going through guides in an interactive environment is a great way to better understand them. See [here](https://jupyter.org/install) for instructions on how to install.\n",
+    "\n",
+    "### Installation\n",
+    "\n",
+    "To install LangChain run:\n",
+    "\n",
+    "```bash npm2yarn\n",
```

Comment:
Make sure this looks good - I thought you have to use the component with how we render markdown from Jupyter

Reply:
Don't we also need to install `@langchain/langgraph` here?

And should we dupe these to LangGraph in general?

Reply:
Can you clarify what you mean by dupe to LangGraph?